c10/util/qint32.h

```diff
@@ -9,6 +9,7 @@ namespace c10 {
 struct alignas(4) qint32 {
   using underlying = int32_t;
   int32_t val_;
+  qint32() = default;
```

Comment:
Why were these constructors added?

Reply:
It seems like CUDALoops(gpu_kernel function mainly) requires default constructor for the result and input data types. I get the following compile errors without default constructors:  
```
pytorch/aten/src/ATen/native/cuda/CUDALoops.cuh(177): error: no default constructor exists for class "c10::qint8"
pytorch/aten/src/ATen/native/cuda/MemoryAccess.cuh(187): error: the default constructor of "at::native::memory::aligned_vector<c10::qint32, 2>" cannot be referenced
```
..... List goes on.  

I am not sure is there any workaround for this(C++ is not my strongest side, suggestions are welcome).